src/Sylius/Behat/Context/Api/Shop/PromotionContext.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Api\Shop;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\Request;
+use Sylius\Behat\Client\ResponseCheckerInterface;
+use Sylius\Behat\Service\SharedStorage;
+use Symfony\Component\HttpFoundation\Request as HttpRequest;
+use Webmozart\Assert\Assert;
+
+final class PromotionContext implements Context
+{
+    /** @var ApiClientInterface */
+    private $cartsClient;
+
+    /** @var SharedStorage */
```

Comment:
```suggestion
    /** @var SharedStorageInterface */
```
and this service could be placed as the last one